src/material-experimental/mdc-slider/slider.ts

```diff
@@ -22,13 +57,220 @@ import {SliderAdapter} from './slider-adapter';
+    }
+  }
+  private _disabled: boolean = false;
+
+  /** Whether the slider displays a numeric value label upon pressing the thumb. */
+  @Input()
+  get discrete(): boolean { return this._discrete; }
+  set discrete(v: boolean) { this._discrete = coerceBooleanProperty(v); }
+  private _discrete: boolean = false;
+
+  /** Whether the slider displays tick marks along the slider track. */
+  @Input()
+  get showTickMarks(): boolean { return this._showTickMarks; }
+  set showTickMarks(v: boolean) {
+    this._showTickMarks = coerceBooleanProperty(v);
+  }
+  private _showTickMarks: boolean = false;
+
+  /** The minimum value that the slider can have. */
+  @Input()
+  get min(): number { return this._min; }
+  set min(v: number) { this._min = coerceNumberProperty(v, this._min); }
+  private _min = 0;
+
+  /** The maximum value that the slider can have. */
+  @Input()
+  get max(): number { return this._max; }
+  set max(v: number) { this._max = coerceNumberProperty(v, this._max); }
+  private _max = 100;
+
+  /** The values at which the thumb will snap. */
+  @Input()
+  get step(): number { return this._step; }
+  set step(v: number) { this._step = coerceNumberProperty(v, this._step); }
+  private _step: number = 1;
+
+  /**
+   * Function that will be used to format the value before it is displayed
+   * in the thumb label. Can be used to format very large number in order
+   * for them to fit into the slider thumb.
+   */
+  @Input() displayWith: ((value: number) => string) | null;
+
+  /** Event emitted when the slider thumb starts being dragged. */
+  @Output() readonly dragStart: EventEmitter<MatSliderDragEvent>
+    = new EventEmitter<MatSliderDragEvent>();
+
+  /** Event emitted when the slider thumb stops being dragged. */
+  @Output() readonly dragEnd: EventEmitter<MatSliderDragEvent>
+    = new EventEmitter<MatSliderDragEvent>();
+
+  /** Instance of the MDC slider foundation for this slider. */
   private _foundation = new MDCSliderFoundation(new SliderAdapter());
+
+  /** Whether the foundation has been initialized. */
+  _initialized: boolean = false;
+
+  /** The string representation of the start thumbs value. */
+  _startValueIndicatorText: string;
+
+  /** The string representation of the end thumbs value. */
+  _endValueIndicatorText: string;
+
+  /** The injected document if available or fallback to the global document reference. */
+  _document: Document;
+
+  /**
+   * The defaultView of the injected document if
+   * available or fallback to global window reference.
+   */
+  _window: Window;
+
+  /** The hosts native HTML element. */
+  _hostElement: HTMLElement;
+
+  /** Used to keep track of & render the active & inactive tick marks on the slider track. */
+  _tickMarks: TickMark[];
+
+  constructor(
+    readonly _cdr: ChangeDetectorRef,
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    private readonly _platform: Platform,
+    @Inject(DOCUMENT) protected readonly document: any) {
```

Comment:
`@Inject(DOCUMENT) doc: any` (no need to save it as a property since you're assigning it to `this._document`)